src/main/java/sg/ncl/MainController.java

```diff
@@ -1929,6 +1929,34 @@ public String stopExperiment(@PathVariable String teamName, @PathVariable String
         return abc(teamName, expId, redirectAttributes, realization, request);
     }
 
+//    @RequestMapping("/get_topology/{teamName}/{expId}")
+//    @ResponseBody
+//    public String getTopology(@PathVariable String teamName, @PathVariable String expId) {
+//        try {
+//            HttpEntity<String> request = createHttpEntityHeaderOnly();
+//            ResponseEntity response = restTemplate.exchange(properties.getTopology(teamName, expId), HttpMethod.GET, request, String.class);
+//            log.info("experiment topo: {}", response.getBody().toString());
+//            return "data:image/png;base64," + response.getBody();
+//        } catch (Exception e) {
+//            log.error("Error getting topology thumbnail", e.getMessage());
+//            return "";
+//        }
+//    }
+
+    @RequestMapping(value = "/get_topology/{teamName}/{expId}", method=RequestMethod.GET, produces = MediaType.IMAGE_PNG_VALUE)
+    @ResponseBody
+    public byte[] getTopology(@PathVariable String teamName, @PathVariable String expId) {
+        try {
+            HttpEntity<String> request = createHttpEntityHeaderOnly();
+            ResponseEntity response = restTemplate.exchange(properties.getTopology(teamName, expId), HttpMethod.GET, request, String.class);
+            log.info("experiment topo: {}", response.getBody().toString());
+            return Base64.decodeBase64(response.getBody().toString());
+        } catch (Exception e) {
+            log.error("Error getting topology thumbnail", e.getMessage());
+            return null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Return an empty array instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1168)